x-pack/plugins/spaces/public/nav_control/solution_view_tour/lib.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { BehaviorSubject, defer, from, of, shareReplay, switchMap } from 'rxjs';
+
+import type { CoreStart } from '@kbn/core/public';
+
+import type { SolutionView } from '../../../common';
+import { SHOW_SPACE_SOLUTION_TOUR_SETTING } from '../../../common/constants';
+import type { SpacesManager } from '../../spaces_manager';
+
+export function initTour(core: CoreStart, spacesManager: SpacesManager) {
+  const showTourUiSettingValue = core.settings.globalClient.get(SHOW_SPACE_SOLUTION_TOUR_SETTING);
+  const hasValueInUiSettings = showTourUiSettingValue !== undefined;
+  const showTour$ = new BehaviorSubject(showTourUiSettingValue ?? true);
+
+  const allSpaces$ = defer(() => from(spacesManager.getSpaces())).pipe(shareReplay(1));
```

Comment:
Well here in this concrete cases the.`initTour`  code is synchronous and Observable solves the problem elegantly (subjective opinion 😊). `allSpaces$` is used in 2 places and caches the result. It is then used in a `switchMap` combined with our `showTour$` Observable.

Reply:
Okay, thanks for adding more details. I’m still slightly concerned about the observable overhead, but if you feel strongly about it, I trust you 🙂

Reply:
I've prevented loading space if the tour global setting is hidden (https://github.com/elastic/kibana/pull/194926/commits/edc1252ba385be9af24ac26f699423ba71ea72f3)

Reply:
I didn't comment your solution. Looking at it it does not seem reactive (this code runs outside React), meaning React will not get updates of the `shouldShowTour()` call. It will call it on component mount and then that's it. Unless I am missing something.